job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/jobs/IvyBuilderContext.groovy

```diff
@@ -0,0 +1,67 @@
+package javaposse.jobdsl.dsl.jobs
+
+import javaposse.jobdsl.dsl.DslContext
+import javaposse.jobdsl.dsl.Item
+import javaposse.jobdsl.dsl.JobManagement
+import javaposse.jobdsl.dsl.RequiresPlugin
+import javaposse.jobdsl.dsl.WithXmlAction
+import javaposse.jobdsl.dsl.helpers.AbstractExtensibleContext
+import javaposse.jobdsl.dsl.helpers.step.AntContext
+
+import static javaposse.jobdsl.dsl.ContextHelper.executeInContext
+
+class IvyBuilderContext extends AbstractExtensibleContext {
+
+    final List<Node> ivyBuilderNodes = []
+    private final List<WithXmlAction> withXmlActions
+
+    IvyBuilderContext(List<WithXmlAction> withXmlActions, JobManagement jobManagement, Item item) {
+        super(jobManagement, item)
+        this.withXmlActions = withXmlActions
+    }
+
+    @Override
+    protected void addExtensionNode(Node node) {
+        ivyBuilderNodes << node
+    }
+
+    void ant(@DslContext(AntContext) Closure antClosure = null) {
+        ant(null, null, null, antClosure)
+    }
+
+    void ant(String targetsStr, @DslContext(AntContext) Closure antClosure = null) {
```

Comment:
Remove the overloaded `ant` methods, the Closure-only variant is enough. Configuring options as method parameters and context methods is confusing and makes the implementation complicate. The overloaded methods have not been removed from `StepContext` to maintain compatibility, but they to not need to be replicated by new contexts.